### DISCUSSION TITLE
Shouldn't directives that take other elements as a value be consistent

### DISCUSSION THREAD

**Discussion by ssang** [alpinejs/alpine#3876] [3. Feature ideas]
With the release of the Anchor plugin, I'm confused as to why this one takes a reference to an HTML element while `x-teleport` takes in a string to pass onto `document.querySelector()`. I feel like since these are used in a similar fashion, the type of value that is passed should be consistent. Are there any plans for the next release to allow `x-teleport` to take an HTML element as the value as that's been requested before?

**Comment by ekwoka** [comment]
I do agree.

It should use a selector, or even a more bespoke connecting of the targets.

Not a great api decision for sure.
